source/binary.cpp

```diff
@@ -310,13 +308,13 @@ spv_result_t Parser::parseInstruction() {
   // has its own logical operands (such as the LocalSize operand for
   // ExecutionMode), or for extended instructions that may have their
   // own operands depending on the selected extended instruction.
-  spv_operand_pattern_t expected_operands(
-      opcode_desc->operandTypes,
-      opcode_desc->operandTypes + opcode_desc->numTypes);
+  _.expected_operands.clear();
+  for (auto i = 0; i < opcode_desc->numTypes; i++)
```

Comment:
I expect i is int and numTypes field is uint16_t and you'll get a compile error.  Recommend doing:

```
auto types_array = &(opcode_desc->operandTypes[0]);
for (unsigned i =0, numTypes = opcode_desc->numTypes ; i < numTypes ; i++)  {
 ... =  types_array[ numTypes - i - 1]
```